client/src/request.js

```diff
@@ -1,38 +1,63 @@
-const photos = (productId) => {
-  return fetch(`http://ec2-3-136-203-39.us-east-2.compute.amazonaws.com:4002/photos/id/${productId}`)
-  .then(res => res.json())
-  .then(productPhotos => {
-    return productPhotos;
-  })
-  .catch(error => {
+const fetchWithTimeout = async (resource, options) => {
+  const { timeout = 4000 } = options;
```

Comment:
I'm not use to this syntax. Could you please explain?

Reply:
So basically this extracts the timeout parameter from the options object that should be passed to the function (see line 1). And setting it equal to 4000 just sets a default in case no timeout is found in the options object. Here's some reference in case you want to deep dive a bit [MDN - Destructuring Assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment )